eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Certs.hs

```diff
@@ -140,17 +164,44 @@ instance
 conwayCertsTransition ::
   forall era.
   ( EraTx era
-  , ShelleyEraTxBody era
+  , ConwayEraTxBody era
+  , ConwayEraPParams era
   , State (EraRule "CERT" era) ~ CertState era
   , Embed (EraRule "CERT" era) (ConwayCERTS era)
-  , Environment (EraRule "CERT" era) ~ DelplEnv era
+  , Environment (EraRule "CERT" era) ~ CertEnv era
   , Signal (EraRule "CERT" era) ~ TxCert era
   ) =>
   TransitionRule (ConwayCERTS era)
 conwayCertsTransition = do
-  TRC (env@(DelegsEnv slot txIx pp tx acnt), certState, certificates) <- judgmentContext
+  TRC
+    ( env@(CertsEnv txIx tx acnt pp slot currentEpoch)
+      , certStateFresh
+      , certificates
+      ) <-
+    judgmentContext
   network <- liftSTS $ asks networkId
 
+  -- Update DRep expiry for all DReps that are voting in this transaction
+  let drepActivity = pp ^. ppDRepActivityL
+      votingDRepCreds =
+        Map.keysSet
+          $ Map.mapMaybeWithKey
+            ( \voter _ -> case voter of
+                DRepVoter cred -> Just cred
+                _ -> Nothing
+            )
```

Comment:
That's what I get for in PR programming. What we want is a Set of just the DRep creds:
```suggestion
        Map.foldlWithKey'
          $ ( \acc voter _ -> case voter of
                DRepVoter cred -> Set.insert cred acc
                _ -> acc
            )
          Set.empty
```

Reply:
Awesome!!! :raised_hands: Super thanks @lehins ! :grin: